components/eventing-controller/pkg/handlers/nats.go

```diff
@@ -90,20 +90,21 @@ func newCloudeventClient(config env.NatsConfig) (cev2.Client, error) {
 
 // SyncSubscription synchronizes the given Kyma subscription to NATS subscription.
 // note: the returned bool should be ignored now. It should act as a marker for changed subscription status.
-func (n *Nats) SyncSubscription(sub *eventingv1alpha1.Subscription, cleaner eventtype.Cleaner, _ ...interface{}) (bool, error) {
+func (n *Nats) SyncSubscription(subscription *eventingv1alpha1.Subscription, cleaner eventtype.Cleaner, _ ...interface{}) (bool, error) {
 	var filters []*eventingv1alpha1.BEBFilter
-	if sub.Spec.Filter != nil {
-		uniqueFilters, err := sub.Spec.Filter.Deduplicate()
+	if subscription.Spec.Filter != nil {
+		uniqueFilters, err := subscription.Spec.Filter.Deduplicate()
 		if err != nil {
 			return false, errors.Wrap(err, "deduplicate subscription filters failed")
 		}
 		filters = uniqueFilters.Filters
 	}
 
 	// Format logger
-	log := utils.LoggerWithSubscription(n.namedLogger(), sub)
+	log := utils.LoggerWithSubscription(n.namedLogger(), subscription)
 
-	subsConfig := eventingv1alpha1.MergeSubsConfigs(sub.Spec.Config, &n.defaultSubsConfig)
+	subscriptionConfig := eventingv1alpha1.MergeSubsConfigs(subscription.Spec.Config, &n.defaultSubsConfig)
+	subscription.Status.CleanEventTypes = make([]string, 0, len(filters))
```

Comment:
why not use `subscription.Status.CleanEventTypes = []string{}`
Or create an array - if you already know the number of filters (if performance is of the essence here) - and access it using the index

Reply:
I think you are right about `[]string{}`.

About the array: I am not sure here, but I think an array needs a constant length when it gets declared, so dynamic 
declaration like `subscription.Status.CleanEventTypes = [len(filters)]string` is illegal. Further, the length of it would need to be declared when `subscription` gets created. Right?